### ISSUE TITLE
Blazor Web App with Identity Razor Components unable to log user out in .NET 8 RC2

### ISSUE THREAD

**Issue by affableashish** [dotnet/aspnetcore#51302]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I installed .NET 8 RC2 and quickly took Identity Razor components for a ride.
I noticed that I can't log out a user clicking the 'Logout' NavLink.

<img width="220" alt="image" src="https://github.com/dotnet/aspnetcore/assets/30603497/ba1f8578-dc5b-462f-8ed3-7330757efb96">

This is the error that appears in the browser when I click it:
`The POST request does not specify which form is being submitted. To fix this, ensure <form> elements have a @formname attribute with any unique value, or pass a FormName parameter if using <EditForm>.`

I checked `LogoutForm.razor` and it indeed has `@formname`.
<img width="550" alt="image" src="https://github.com/dotnet/aspnetcore/assets/30603497/3787e1b6-79e3-4d7c-ac5b-b0635ec201dc">

### Also: 
There were some required services that were not added by the template. I have documented them [here](https://github.com/affableashish/blazor-identity-api).
Also there were some weird errors showing which didn't fail the build (so could be a Rider issue).

<img width="1405" alt="image" src="https://github.com/affableashish/blazor-api-identity/assets/30603497/bda6aae4-1bf8-4e3a-b31b-d05dba1038f0">

### Expected Behavior

Clicking 'Logout' should log the user out.

### Steps To Reproduce

1. Clone this repo: https://github.com/affableashish/blazor-identity-api
2. Run the [BlazorServerClient](https://github.com/affableashish/blazor-identity-api/tree/main/BlazorServerClient) project, login using Username: `ashish@example.com` and Password: `Password123!` or any account (it doesn't matter).
3. Try to Log out

### Exceptions (if any)

`The POST request does not specify which form is being submitted. To fix this, ensure <form> elements have a @formname attribute with any unique value, or pass a FormName parameter if using <EditForm>.`

### .NET Version

8.0.100-rc.2.23502.2

**Comment by NitroDH** [comment]
Having the same problem over here made a few attempts to troubleshoot it deeper but no luck yet.

**Comment by NitroDH** [comment]
Alright, found several conversations on this:

https://github.com/dotnet/aspnetcore/issues/50882

https://github.com/dotnet/aspnetcore/issues/51180

with a fix merged here:

https://github.com/dotnet/razor/pull/9324

Hosed until GA unless you wanted to utilized Daily RTM Builds most likely.

**Comment by mkArtakMSFT** [comment]
Thanks for reaching out.
@NitroDH is right, this has been addressed already and will be included in the 8.0 GA release.

**Comment by 333fred** [comment]
Thanks all for your patience while we got a workaround for this issue ready to go. We've published a [toolset razor compiler package](https://www.nuget.org/packages/Microsoft.Net.Compilers.Razor.Toolset/7.0.0-preview.23512.5) that can be installed to fix the build from the command line. Add this reference to the csproj of your project:

```xml
<PackageReference Include="Microsoft.Net.Compilers.Razor.Toolset" Version="7.0.0-preview.23512.5">
  <PrivateAssets>all</PrivateAssets>
  <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
</PackageReference>
```

Make sure to remove this package after you upgrade to the next version of .NET 8: it's only needed for RC2 due to our fixes for this issue not quite making the final build.
